flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointFailureManager.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * The checkpoint failure manager which centralized manage checkpoint failure processing logic.
+ */
+public class CheckpointFailureManager {
+
+	private final static int MAXIMUM_TOLERABLE_FAILURE_NUMBER = Integer.MAX_VALUE;
+
+	private final AtomicInteger continuousFailureCounter;
+	private final int tolerableCpFailureNumber;
+	private final FailureHandlerCallback failureCallback;
+
+	public CheckpointFailureManager(int tolerableCpFailureNumber, FailureHandlerCallback failureCallback) {
```

Comment:
So from our previous discussion, it seems like `FailureHandlerCallback failureCallback` was moved from `handleCheckpointException` to the constructor. It also looks like the only purpose that we will use this is providing a method to fail the job to the `CheckpointFailureManager`. Maybe we should then also just document it that way, and this does not truly justify a concept by itself and could just be `Runnable` instead? I mean we would give a class or even just variable name, wouldn't something along `failJobCallback` make more sense?